test/functional/neutronless/disconnected_service/test_disconnected_service_creation.py

```diff
@@ -608,6 +588,39 @@ def create_mock_rpc_plugin():
 
     assert post_destroy_uris == empty_set
 
+def test_featureoff_grm_listener(setup_l2adjacent_test, bigip):
+    def create_mock_rpc_plugin():
+        mock_rpc_plugin = mock.MagicMock(name='mock_rpc_plugin')
+        mock_rpc_plugin.get_port_by_name.return_value = [
+            {'fixed_ips': [{'ip_address': '10.2.2.134'}]}
+        ]
+        mock_rpc_plugin.get_all_loadbalancers.return_value = [
+            {'lb_id': u'50c5d54a-5a9e-4a80-9e74-8400a461a077',
+             'tenant_id': u'128a63ef33bc4cf891d684fad58e7f2d'
+            }
+        ]
+        return mock_rpc_plugin
+
+    icontroldriver, start_registry = handle_init_registry(bigip,
+                                                          FEATURE_OFF_GRM,
+                                                          create_mock_rpc_plugin)
+
+    service = deepcopy(SEGID_CREATELISTENER)
+    logcall(setup_l2adjacent_test,
+            icontroldriver._common_service_handler,
+            service)
+    after_create_registry = register_device(bigip)
+    empty_set = set()
+
+    create_uris = (set(after_create_registry.keys()) -
+                   set(start_registry.keys()))
+    assert create_uris == (SEG_INDEPENDENT_LB_URIS_GRM | NOSEG_LB_URIS |
+                           NOSEG_LISTENER_URIS)
+
+    logfilename = setup_l2adjacent_test.baseFilename
```

Comment:
Instead of logging to a file and reading from a file, we can also patch the logger throughout the length of the test. It allows for easy inspection of the log, yet it also replaces a real logger with a patched version. There are trade-offs. Not necessary by any means.